### PR TITLE
Document frontmatter's id tag

### LINKED ISSUE #477: Definition for frontmatter id tags
As mentioned at https://github.com/tc39/test262/issues/472#issuecomment-173006824:

> @littledan: Rather than IDs which point to specific spec sections, I think it'd be much more useful to have ids that are the anchors within the document. These are much more likely to be consistent across releases. You could call these esids.

Based on this, and as it's easy to locate the section ids on tc39/ecma262 on each of the `emu-clause` tags, which are expected to be unique, I believe we can use the respective hash id for the Frontmatter's `id` and further versions:

```yaml
id: sec-completion-record-specification-type
```

this value can be converted to links that could be found anywhere:

- https://tc39.github.io/ecma262/#sec-completion-record-specification-type
- http://www.ecma-international.org/ecma-262/6.0/index.html#sec-constructor-properties-of-the-global-object
- mylocalcopy#sec-constructor-properties-of-the-global-object

This reference works good and it's still compatible with the previous formats, as won't conflict with the recently adopted `es7id: pending` value, which can be easily replaced by `id: pending` and it's still useful for anything on Stage 3 and it's not on the draft yet.

@goyakin, do you like this idea?

### PR DESCRIPTION
**Pull request by leobalter**
Closes #477

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,6 +90,13 @@ When writing a new test for ES6, it is only necessary to include this tag when t
 
 This tag identifies the portion of the ECMAScript 6 standard that is tested by this test.
```
Comment (goyakin) [inline CONTRIBUTING.md]: The new `id` is an anchor in the spec whereas the old ones are section numbers. I think we should update the old ones' descriptions to call out the difference.

Comment (leobalter) [inline CONTRIBUTING.md]: :+1: I will do it

Comment (leobalter) [inline CONTRIBUTING.md]: 8375822

@goyakin tell me if it needs improvement.

**Comment by goyakin** [comment]
Merged as cb754d2450d682afc25c6d263cb2a7ff17862a7d

**Pull request closed without merging**